### DISCUSSION TITLE
Different autofit behavior between simple table and flextable

### DISCUSSION THREAD

**Discussion by gilleschapron** [davidgohel/officer#552] [Q&A] [answered]
Hi @davidgohel, hope your well,

I have to print table with very long concatenated strings, and the behavior to calculate width of columns between simple table in officer and using flextable is very different.

Example : 
```r
library(officer)
library(flextable)

temp <- data.frame(colonne_1 = c("1", "2"),
                   colonne_2 = paste0(rep("toto", 50), collapse = ", "))

doc <- read_docx()

# simple table
doc <- body_add_table(doc, temp)
doc <- body_add_par(doc, "")

# flextable
tempft <- flextable(temp)
doc <- body_add_flextable(doc, tempft)
doc <- body_add_par(doc, "")

# with autofit
tempft2 <- flextable(temp)
tempft2 <- autofit(tempft2)
doc <- body_add_flextable(doc, tempft2)

print(doc, "my_doc.docx")
```

Autofit for simple table is perfect.
Flextable without autofit has very tiny widths.
Flextable with autofit has to large widths.

Do I miss something ? Is there an option to apply autofit behavior of simple tables into flextables ?

**Comment by davidgohel** [comment, marked as answer]
Hello Gilles,

This should set real Word autofit: 

```
myft <- set_table_properties(myft, layout = "autofit")
```

PS : sorry for the naming issue, it's historical... 

- `autofit()` set the width to the calculated width of the max paragraph length (and only `\n` can be used otherwise it is considered to be layout on a single row), 
- `set_table_properties(layout = "autofit")` dedicate the autofit work to the sofware that will be used to view the table (browser for HTML, adobe for PDF, Word for docx)

More here: https://ardata-fr.github.io/flextable-book/layout.html#tabwidth

**Reply by gilleschapron** [reply]
Thank you!
Your right, I test that, but like in your link, like a `set` function without assign it to the flextable.  
If I do : 
```r
ft <- set_table_properties(ft, layout = "autofit")
```
It works.
But not if I do this:
 ```r
ft
set_table_properties(ft, layout = "autofit")
```
Nevertheless, syntax is correct here : 
https://davidgohel.github.io/flextable/reference/set_table_properties.html

**Reply by davidgohel** [reply]
I don't get the `set_` remark. I never wrote object need not be assigned. `setNames()` do need assignments, so do `rlang::set_names()` and many other functions. I suspect you are a data.table user ;)

**Reply by gilleschapron** [reply]
Yes exactly I thought to data.table `set` functions seeing:
![image](https://github.com/davidgohel/officer/assets/10176335/c69c667b-a055-4511-8eea-b8373ee7f702)
in https://ardata-fr.github.io/flextable-book/layout.html#tabwidth
:-)

**Reply by davidgohel** [reply]
You misunderstood this code, when no assignment is made, the flextable is returned and the print method is displayed, that's why you can see the result table below :) A print method is often associated to object (think about how you print data.frame in the console by just typing the name of the table)

Flextable functions return the flextable, this enable piping (as with dplyr verbs or `DT[, list(x = mean(z)]`):

```
mtcars %>%
  flextable() %>% theme_box() %>%
  set_table_properties(layout = "autofit")
```

**Reply by gilleschapron** [reply]
Humm, seems it's friday evening, and I need some rest :-D
Sorry, and thank you for your patience...

**Reply by davidgohel** [reply]
no problem, have a good week end
